Server/src/server/gameCenter/models/game/Player.java

```diff
@@ -91,6 +92,35 @@ public void addCardToDeck(ServerCard card) throws LogicException {
         return drawnCards;
     }
 
+    public ServerCard[] getCardsFromDeckNoDuplicate(int cardsToDraw, ServerCard card) {
+        ServerCard[] drawnCards = new ServerCard[cardsToDraw];
+        Random RNGenerator = new Random();
+        int infiniteRedrawFailSafe = 10;
+
+		for(int i = 0; i < cardsToDraw; i++) {
+            if (!deck.getCards().isEmpty()) {
+                int index = RNGenerator.nextInt(deck.getCards().size());
+                ServerCard drawnCard = deck.getCards().get(index);
+                if( drawnCard.IDSameAs(card.getCardId()) && infiniteRedrawFailSafe > 0){
+                    System.out.println("Same card found, redrawing...");
+                    infiniteRedrawFailSafe--;
+                    i--;
+                    continue;
+                }
+                drawnCards[i] = drawnCard;
+                infiniteRedrawFailSafe = 10;
+                try {
+                    deck.removeCard(drawnCard);
+                } catch (ClientException ignored) {
+                    System.out.println("Unable to remove card from deck");
```

Comment:
For debugging, this error would be more helpful if it where more specific.  E.g add a reference to the card we could not remove.  (maybe its null -- if it is, I'd like to know)